test/test_literal/test_literal.py

```diff
@@ -80,6 +83,60 @@ def test_cant_pass_invalid_lang(
         with pytest.raises(exception_type):
             Literal("foo", lang=lang)
 
+    @pytest.mark.parametrize(
+        "lexical, datatype, is_ill_formed",
+        [
+            ("true", XSD.boolean, False),
+            ("1", XSD.boolean, False),
+            (b"false", XSD.boolean, False),
+            (b"0", XSD.boolean, False),
+            ("yes", XSD.boolean, True),
+            ("200", XSD.byte, True),
+            (b"-128", XSD.byte, False),
+            ("127", XSD.byte, False),
+            ("255", XSD.unsignedByte, False),
+            ("-100", XSD.unsignedByte, True),
+            (b"200", XSD.unsignedByte, False),
+            (b"64300", XSD.short, True),
+            ("-6000", XSD.short, False),
+            ("1000000", XSD.nonNegativeInteger, False),
+            ("-100", XSD.nonNegativeInteger, True),
+            ("a", XSD.double, True),
+            ("0", XSD.double, False),
+            ("0.1", XSD.double, False),
+            ("0.1", XSD.decimal, False),
+            ("0.g", XSD.decimal, True),
+            ("b", XSD.integer, True),
+            ("2147483647", XSD.int, False),
+            ("2147483648", XSD.int, True),
+            ("2147483648", XSD.integer, False),
+            ("valid ASCII", XSD.string, False),
```

Comment:
I promise this is only 50% a digression:

Is there any temptation to try non-ASCII strings within this test series?  Or does the waiver embedded in the `pytest` option [`disable_test_id_escaping_and_forfeit_all_rights_to_community_support`](https://docs.pytest.org/en/6.2.x/parametrize.html) kinda nix that plan?

In more seriousness, non-ASCII data handling has taken a non-trivial amount of attention from several communities I've worked with, and it's always nice to see handling guidance encoded.

Reply:
I'm happy to include non ascii-strings, we can easily set the test ID manually if we don't like the inferred test ID [[ref](https://docs.pytest.org/en/6.2.x/reference.html#pytest-param)] if that is what you mean. If you have any ideas please do add them.

Reply:
I will google for some phrases from different languages and add them.

Reply:
I've encountered a few different pain points:
* Purported support for unicode that could not withstand the appearance of the unicode snowman: ☃.
* Different languages being transcoded to UTF-8, but with an incorrect guess at the original encoding (IIRC, Latin-accented characters guessed to be ISO-8859-1 but being something else).
* UTF-16BE being dumped as ASCII without transcoding, so every other byte is `\x00`.

Reply:
Also, truly, I defer to you on the best way to handle `parameterize`.  I'm a bit too new to that method to have any ideas about best practices.

Reply:
I will include the snowman, and some non latin script phrases, probably we should have more extensive testing for this but it is easy to include here.

Reply:
Added a string with arabic script and a snowman, but this does not do much to test this handling.